contracts/oracle/BackedOracle.sol

```diff
@@ -0,0 +1,201 @@
+        require(_latestRoundNumber != 0, "No data present");
+
+        return _roundData[_latestRoundNumber].answer;
+    }
+
+    function latestTimestamp() external view override returns (uint256) {
+        require(_latestRoundNumber != 0, "No data present");
+
+        return _roundData[_latestRoundNumber].timestamp;
+    }
+
+    function latestRound() external view override returns (uint256) {
+        require(_latestRoundNumber != 0, "No data present");
+
+        return _latestRoundNumber;
+    }
+
+    function latestRoundData()
+        external
+        view
+        override
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        require(_latestRoundNumber != 0, "No data present");
+
+        return (
+            uint80(_latestRoundNumber),
+            _roundData[_latestRoundNumber].answer,
+            _roundData[_latestRoundNumber].timestamp,
+            _roundData[_latestRoundNumber].timestamp,
+            uint80(_latestRoundNumber)
+        );
+    }
+
+    function getAnswer(
+        uint256 roundId
+    ) external view override returns (int256) {
+        require(roundId <= _latestRoundNumber, "No data present");
+
+        return _roundData[roundId].answer;
+    }
+
+    function getTimestamp(
+        uint256 roundId
+    ) external view override returns (uint256) {
+        require(roundId <= _latestRoundNumber, "No data present");
+
+        return _roundData[roundId].timestamp;
+    }
+
+    function getRoundData(
+        uint80 roundId
+    )
+        external
+        view
+        override
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        require(roundId <= _latestRoundNumber, "No data present");
+
+        return (
+            roundId,
+            _roundData[roundId].answer,
+            _roundData[roundId].timestamp,
+            _roundData[roundId].timestamp,
+            roundId
+        );
+    }
+
+    function updateAnswer(
+        int192 newAnswer,
+        uint32 newTimestamp
+    ) external onlyRole(UPDATER_ROLE) {
+        int256 latestAnswer = _roundData[_latestRoundNumber].answer;
+        uint256 latestTimestamp = _roundData[_latestRoundNumber].timestamp;
+
+        // Timestamp is actual timestamp
+        require(
+            newTimestamp < block.timestamp,
+            "Timestamp cannot be in the future"
+        );
+
+        // Check that the timestamp is not too old
+        require(
+            block.timestamp - newTimestamp <= MAX_TIMESTAMP_AGE,
+            "Timestamp is too old"
+        );
+
+        // The timestamp is more than the last timestamp
+        require(
+            newTimestamp > latestTimestamp,
+            "Timestamp is older than the last update"
+        );
+
+        // The last update happened more than MIN_UPDATE_INTERVAL ago
+        require(
+            newTimestamp - latestTimestamp > MIN_UPDATE_INTERVAL,
+            "Timestamp cannot be updated too often"
```

Comment:
more like "Answer cannot be updated too often"

Reply:
Acknowledged. Leaving as is.